### DISCUSSION TITLE
Strange behavior when mapping properties with default values

### DISCUSSION THREAD

**Discussion by art-grig** [nartc/mapper#383] [Q&A]
Hi!
Let's assume we have following property both in model and entity:

```ts
@AutoMap()
isActive = true;
```

If not doing anything special and trying to map for example model to entity, this prop is going to be mapped as anonymous `object`...
Which is really strange behavior as for me.

The only way for fixing this issue is specifying the `typeFn` explicitly on both sides:

```ts
@AutoMap({ typeFn: () => Boolean })
isActive = true;
```

Is there any better solution for this issue?
And perhaps it makes sense to throw an exception on mapper configuration with clear description?

**Comment by nartc** [comment]
Unfortunately, this is TypeScript limitation :(. Reflection cannot infer the correct type for this default value. This is probably in conflict with some Lint rules. I'd say that `typeFn` is the best workaround as of the moment.
